packages/server/movement.ts

```diff
@@ -0,0 +1,242 @@
+import {
+    GameMap, Game, PlayerIndex, Unit, UnitId, Component, Position, TilePos, UnitState, PresenceMap,
+    Hp, Mover, Attacker, Harvester, ProductionFacility, Builder, Vision,
+    Action, ActionFollow, ActionAttack,
+} from './types';
+
+import * as V from './vector.js'
+import { notEmpty } from './tsutil.js'
+
+export function checkMovePossibility(unit: Unit, gm: GameMap, presence: PresenceMap): Position {
+    const currentPos = unit.position;
+    unit.debug ??= {};
+
+    const explode = (p: TilePos) => p.x+p.y*gm.w; 
+
+    // Disable collisions for harvesting units
+    if (unit.actionQueue.length > 0 &&
+        unit.actionQueue[0].typ === 'Harvest'
+    ) {
+        let velocity = {
+            x: Math.cos(unit.direction) * 1,
+            y: -Math.sin(unit.direction) * 1

```

Comment:
`* 1` - hmm, I'm not sure if this adds any value here

Reply:
Same as before